### ISSUE TITLE
GitHub: deprecation of using access_token query param

### ISSUE THREAD

**Issue by fsworld009** [python-social-auth/social-core#430]
GitHub announced deprecation of using `access_token` query param to make API requests:

https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api/

This package's github backend needs to migrate to use [Basic Authentication](https://developer.github.com/v3/auth/#via-oauth-and-personal-access-tokens) in order to continue supporting GitHub.

**Comment by fsworld009** [comment]
Saw people already reported

**Comment by fsworld009** [comment]
Just saw someone already made a PR for it https://github.com/python-social-auth/social-core/pull/428

**Comment by asherf** [comment]
Thanks for creating the issue !
IMHO we should keep this issue open until the PR is merged!

**Comment by rafaelcapucho** [comment]
Hi, I'm facing the same issue, receiving these emails:

```
Hi @rafaelcapucho,

On February 9th, 2020 at 22:37 (UTC) your application (OauthGitHubStudentDeveloperPack) used an access token (with the User-Agent python-requests/2.22.0) as part of a query parameter to access an endpoint through the GitHub API:

https://api.github.com/user

Please use the Authorization HTTP header instead as using the `access_token` query parameter is deprecated and will be removed July 1st, 2020.

Depending on your API usage, we'll be sending you this email reminder once every 3 days.

Visit https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api/#authenticating-using-query-parameters for more information.

Thanks,
The GitHub Team
```

Thank you

**Comment by vsoch** [comment]
Me as well!

**Comment by mateuszmandera** [comment]
@fsworld009 @omab This can be closed, this was a duplicate of another issue on this, the https://github.com/kiwitcms/github-app/pull/22 PR is merged.

**Comment by omab** [comment]
Fixed by #428
